src/git/repo.rs

```diff
@@ -965,6 +1022,144 @@ Either create it, or update the main branch setting by running:
             .map_err(wrap_git_error)?;
         Ok(make_non_zero_oid(oid))
     }
+
+    /// Amends the provided parent commit in memory and returns the resulting tree.
+    ///
+    /// Only amends the files provided in the options, and only supports amending from
+    /// either the working tree or the index, but not both.
+    ///
+    /// See `Repo::cherry_pick_fast` for motivation for performing the operation
+    /// in-memory.
+    pub fn amend_fast(
+        &self,
+        parent_commit: &Commit,
+        opts: &AmendFastOptions,
+    ) -> eyre::Result<Tree> {
+        let parent_commit_pathbufs = self
+            .get_paths_touched_by_commit(parent_commit)?
+            .ok_or_else(|| {
+                eyre::eyre!(
+                    "Could not get paths touched by commit: {:?}",
+                    &parent_commit
+                )
+            })?
+            .into_iter()
+            .collect_vec();
+        let changed_paths = {
```

Comment:
Since you're not using `parent_commit_pathbufs` elsewhere in this function, you can collect it directly into a `HashSet<PathBuf>` and then add the elements and convert it into a `Vec`. (Actually, the interface to `dehydrate_commit` could be refactored, because I think all of the call-sites end up taking `HashSet`s anyways.